### ISSUE TITLE
[ENH]: User-defined named colors

### ISSUE THREAD

**Issue by AlvaroFurlani** [matplotlib/matplotlib#23967]
### Problem

Would it be possible to add a functionality that allows users to define their own named colors? It would be very helpful for cases in which visualizations need to comply with corporation brand requirements. Of course, users could specify the desired color code (such as rgb or hex) in each plot, but having named colors is more convenient.

### Proposed solution

This is a temporary solution that I came up with: https://stackoverflow.com/questions/73423618/registering-named-colors-in-matplotlib.

But as mentioned in the stackoverflow post, it is far from ideal.

**Comment by tacaswell** [comment]
Do you need to use the colors someplace that only takes strings (the only place I can think of that require strings is rcparams or style files where it is probably OK to put the HEX/RGB)?

If not, I would suggest doing something like


```python
from types import SimpleNamespace
my_colors = SimpleNamespace(red='hex', blue='hex', ...)

...
ax.plot(..., color=my_colors.red)
ax.plot(..., color=my_colors.blue)
```

If this is to match a company style, put this is a package you can ensure is installed in your environments.


-----


Please do not edit the installed files from any package, it is essentially doing an undocumented fork!  If you have control of the package index you install from and really want to do it that way, I would patch as part of the build process.

**Comment by timhoffm** [comment]
@AlvaroFurlani can you describe the use case?

I can see an argument for named color sequences (similar to what seaborn calls *color palette*, because there you name a more complex construct.

However for single colors,

- I don't think this should be used globally, because name clashes are likely; e.g. multiple libraries may want to define 'navyblue'.
- In your local script
  ```
  navy_blue = '#3333ff'
  plt.plot(x, y, color=navy_blue)
  ```
  is equally good if not better (simpler) than
  ```
  register_color('navy_blue', '#3333ff')
  plt.plot(x, y, color='navy_blue')
  ```

**Comment by AlvaroFurlani** [comment]
@timhoffm my main interest in having new named colors is to comply with corporate brand requirements. For example, my current company defines a set of 17 hues with 10 tones each (so 170 total colors). Using their hex codes in each plotting function is not very efficient, as I would have to go check them in a table every time. On the other hand, named colors is more intuitive and easier to remember.

My initial solution was to have a separate file with a dict of colors definitions (similar to what @story645 mentioned above) and import it in every file where I used matplotlib. In that dict, I used colors names such as 'xxx-blue-100', 'xxx-blue-200', 'xxx-blue-300', and so on..., where 'xxx' is the abbreviation of my current company name (so there is no risk of name conflicts with other color names) and the numbers represent the color tone.

While that solution works, I think it would be easier to register the new color names once and be able to use them anytime in the future without having to import a dict of color definitions from a separate file every time.

**Comment by jklymak** [comment]
How is registering the names superior to importing a package to define a dict?  They both require an import, don't they?

**Comment by AlvaroFurlani** [comment]
@jklymak Not necessarily. Currently, the module matplotlib.colors contains various dicts with the named colors (BASE_COLORS, CSS4_COLORS, TABLEAU_COLORS...) and you don't need an import to use them. So the registering process could create new dicts in matplotlib.colors, so that any user-defined named colors are readily available.

**Comment by jklymak** [comment]
Registering wouldn't write into `matplotlib.colors`.  You would need to do that at start-up.  See our registering of colormaps.

**Comment by oscargus** [comment]
I agree that it would be nice to be able to do this from some sort of settings file (maybe not rcParams). Here is a concrete example how I solved it for my university: https://gitlab.liu.se/da/liu-colors/-/blob/master/matplotlib/liucolors.py
But if a sys-admin could make sure that those colors are installed and can be accessed using the name directly, rather than using the dict that would be quite useful.

(The `SimpleNamespace` solution is clearly better than a dict since one can tab-complete and I will probably adapt it in the near future.)

(I should also say that I did this long before actually starting to know Matplotlib a bit better so I acknowledge that some of the code can be written in a better way.)
